files/default/dcv/pcluster_dcv_connect.sh

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env bash
+#
+# Cookbook Name:: aws-parallelcluster
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script must be executed by the user who want to connect to DCV to obtain a Session Token.
+#
+# It performs the following steps:
+# 1. Asks the ParallelCluster DCV External Authenticator for a Request Token and the Access File name
+# 2. Creates the "access file" in the AUTHORIZATION_FILE_DIR folder (to confirm user identity)
+# 3. Asks the ParallelCluster DCV External Authenticator for a SessionToken (the real token to access to the DCV session)
+# 4. Returns DCV Session Id, DCV Server Port and the Session Token
+
+# Requirements:
+# jq, curl, awk, xargs, shuf, cat, grep, systemctl
+
+# Usage:
+# The script requires the ParallelCluster shared folder as input parameter
+# The ParallelCluster shared folder will be used as session storage folder for DCV.
+# It enables the clients to share files while connected to the session.
+# ./pcluster_dcv_connect.sh "/shared"
+
+
+# Returns the sessionid, the port and the tokenid (256 character long).
+# Example: mysession 8443 adfsaklcxzvsadkhfgsdkhjfag-__bafbdajshsdjfh
+
+AUTHORIZATION_FILE_DIR="/var/spool/parallelcluster/pcluster_dcv_authenticator"
+DCV_SESSION_FOLDER="${HOME}/.parallelcluster/dcv"
+LOG_FILE_PATH="/var/log/parallelcluster/pcluster_dcv_connect.log"
+LOG_FILE_MAX_SIZE=5242880 # 5MB
+
+
+_fail() {
+  message=$1
+  >&2 echo "ERROR: ${message}"
+  exit 1
+}
+
+_check_if_empty() {
+  variable=$1
+  message=$2
+  if [[ -z "${variable}" ]]; then
+      _fail "${message}"
+  fi
+}
+
+_atomic_touch() {
+    file_path="$1"
+
+    _umask=$(umask)
+    umask 0022 && touch "${file_path}"
+    umask ${_umask}
+
+    if [[ ! -f "${file_path}" ]]; then
+        _fail "Unable to create the file in the $(dirname "${file_path}") folder. Exiting.."
+    fi
+}
+
+_log() {
+    text=$1
+
+    log_time=$(date "+%Y-%m-%d %H:%M:%S")
+
+    file_size=$(du -b ${LOG_FILE_PATH} | awk '{print $1}')
+
+    [[ -f "${LOG_FILE_PATH}" ]] || _atomic_touch "${LOG_FILE_PATH}"
+
+    if [[ ${file_size} -gt ${LOG_FILE_MAX_SIZE} ]]; then
+        mv "${LOG_FILE_PATH}" "${LOG_FILE_PATH}.1"
+        echo "[${log_time}]: ${text}" > "${LOG_FILE_PATH}"
```

Comment:
once you move the file, you can do the append >> in place of the >. 
This will factor out with the line in the else case, hence the else could be removed